src/Razor/src/Microsoft.AspNetCore.Razor.OmniSharpPlugin/BackgroundDocumentProcessedPublisher.cs

```diff
@@ -0,0 +1,286 @@
+            {
+                // Today we only support publishing Razor component files to the workspace. We could choose to publish more but we're being
+                // purposefuly restrictive.
+                return;
+            }
+
+            var openVirtualFilePath = document.FilePath + ActiveVirtualDocumentSuffix;
+            var openDocument = _workspace.GetDocument(openVirtualFilePath);
+            if (openDocument != null)
+            {
+                // This document is open in the editor, no reason for us to populate anything in the workspace the editor will do that.
+                return;
+            }
+
+            var backgroundVirtualFilePath = document.FilePath + BackgroundVirtualDocumentSuffix;
+            var currentDocument = _workspace.GetDocument(backgroundVirtualFilePath);
+            if (currentDocument == null)
+            {
+                // Background document doesn't exist, we need to create it
+
+                var roslynProject = GetRoslynProject(document.Project);
+                if (roslynProject == null)
+                {
+                    // There's no Roslyn project associated with the Razor document.
+                    _logger.LogTrace($"Could not find a Roslyn project for Razor virtual document '{backgroundVirtualFilePath}'.");
+                    return;
+                }
+
+                var documentId = DocumentId.CreateNewId(roslynProject.Id);
+                var name = Path.GetFileName(backgroundVirtualFilePath);
+                var emptyTextLoader = new EmptyTextLoader(backgroundVirtualFilePath);
+                var documentInfo = DocumentInfo.Create(documentId, name, filePath: backgroundVirtualFilePath, loader: emptyTextLoader);
+                _workspace.AddDocument(documentInfo);
+                currentDocument = _workspace.GetDocument(backgroundVirtualFilePath);
+
+                Debug.Assert(currentDocument != null, "We just added the document, it should definitely be there.");
+            }
+
+            // Update document content
+
+            var sourceText = document.GetGeneratedCodeSourceText();
+            _workspace.OnDocumentChanged(currentDocument.Id, sourceText);
+        }
+
+        public override void Initialize(OmniSharpProjectSnapshotManager projectManager)
+        {
+            if (projectManager is null)
+            {
+                throw new ArgumentNullException(nameof(projectManager));
+            }
+
+            _projectManager = projectManager;
+            _projectManager.Changed += ProjectManager_Changed;
+        }
+
+        // Here we're specifically listening for cases when a user opens a Razor document and an active C# content gets created.
+        internal void Workspace_WorkspaceChanged(object sender, WorkspaceChangeEventArgs args)
+        {
+            switch (args.Kind)
+            {
+                case WorkspaceChangeKind.DocumentChanged:
+                    {
```

Comment:
Indentation

Reply:
This is actually the default implementation imposed by C# 😄